src/blueprints/tv2/action-factories/tv2-graphics-action-factory.ts

```diff
@@ -164,4 +208,319 @@ export class Tv2GraphicsActionFactory {
+      type: PartActionType.INSERT_PART_AS_NEXT,
+      data: {
+        partInterface: partInterface,
+        pieceInterfaces: [pieceInterface]
+      },
+      metadata: {
+        contentType: Tv2ActionContentType.GRAPHICS
+      }
+    }
+  }
+
+  private createVizFullscreenGraphicsPieceInterface(partId: string, blueprintConfiguration: Tv2BlueprintConfiguration, fullscreenGraphicsData: Tv2FullscreenGraphicsManifestData): Tv2PieceInterface {
+    return this.createGraphicsPieceInterface({
+      id: 'fullscreenGraphicsPiece',
+      partId,
+      name: fullscreenGraphicsData.name,
+      preRollDuration: blueprintConfiguration.studio.VizPilotGraphics.PrerollDuration,
+      pieceLifespan: this.findPieceLifespan(blueprintConfiguration, fullscreenGraphicsData.name),
+      layer: Tv2SourceLayer.PILOT_GRAPHICS,
+      content: {
+        fileName: `${PILOT_PREFIX}${fullscreenGraphicsData.vcpId}`,
+        path: `${fullscreenGraphicsData.vcpId}`,
+      },
+      timelineObjects: this.createVizFullscreenPilotGraphicsTimelineObjects(blueprintConfiguration, fullscreenGraphicsData),
+      metadata: {
+        type: Tv2PieceType.GRAPHICS,
+        outputLayer: Tv2OutputLayer.PROGRAM
+      }
+    })
+  }
+
+  private createGraphicsPartInterface(partInterfaceWithRequiredValues: Pick<PartInterface, 'id' | 'name'> & Partial<PartInterface>): PartInterface {
+    return {
+      disableNextInTransition: false,
+      expectedDuration: 0,
+      inTransition: {
+        keepPreviousPartAliveDuration: 0,
+        delayPiecesDuration: 0
+      },
+      isNext: false,
+      isOnAir: false,
+      isPlanned: false,
+      outTransition: {
+        keepAliveDuration: 0
+      },
+      pieces: [],
+      rank: 0,
+      segmentId: '',
+      isUnsynced: false,
+      ...partInterfaceWithRequiredValues
+    }
+  }
+
+  private findPieceLifespan(blueprintConfiguration: Tv2BlueprintConfiguration, templateName: string): PieceLifespan {
+    const template: GraphicsTemplate | undefined = blueprintConfiguration.showStyle.graphicsTemplates.find(
+      graphic => graphic.vizTemplate ? graphic.vizTemplate.toUpperCase() === templateName.toUpperCase() : false
+    )
+    const templateOutType: TemplateOutType = this.parseTemplateOutType(template?.outType ?? '')
+    return this.getPieceLifespanFromOutType(templateOutType)
+  }
+
+  private parseTemplateOutType(template: string): TemplateOutType {
+    switch (template) {
+      case 'S':
+        return TemplateOutType.S
+      case 'O':
+        return TemplateOutType.O
+      case 'B':
+      default:
+        return TemplateOutType.B
+    }
+  }
+
+  private getPieceLifespanFromOutType(outType: TemplateOutType): PieceLifespan {
+    switch (outType) {
+      case TemplateOutType.B:
+        return PieceLifespan.WITHIN_PART
+      case TemplateOutType.S:
+        return PieceLifespan.SPANNING_UNTIL_SEGMENT_END
+      case TemplateOutType.O:
+        return PieceLifespan.STICKY_UNTIL_RUNDOWN_CHANGE
+    }
+  }
```

Comment:
It was made together with Anders, and eliminated the `if` statement which looked weird and was somewhat difficult to understand.

Reply:
But you only use it here and you only use it for getting the `PieceLifespan`, so here, you might as well just return the `PieceLifspan` directly instead of making a second function for it.

I am introducing a `StudioBlueprintConfigurationMapper`. In there I'll map to the `TemplateOutType` so we only need one switch here.

Reply:
Souns like i shouldn't make any more change to this then, unless the PR with the `StudioBlueprintConfigurationMapper` is merged before this.

Reply:
It won't be. It is dependent on this PR